jupyter_collaboration/rooms.py

```diff
@@ -212,49 +219,60 @@ def _on_document_change(self, target: str, event: Any) -> None:
             document. This tasks are debounced (60 seconds by default) so we
             need to cancel previous tasks before creating a new one.
         """
-        if self._update_lock.locked():
-            return
-
-        self._saving_document = asyncio.create_task(
-            self._maybe_save_document(self._saving_document)
-        )
+        asyncio.create_task(self._maybe_save_document())
```

Comment:
Always [save a reference to the task](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task).

Reply:
Thanks for this callout! I was not aware of this detail.

Reply:
Fixed in latest revision.